### ISSUE TITLE
macro requires in module macro

### ISSUE THREAD

**Issue by Kethku** [Olical/aniseed#5]
Would it make sense to somehow add macro import support to the module macro? It would be awesome to just add a tag to an import banding or something similar to trigger importing macros at the same time.

**Comment by Olical** [comment]
The "require" syntax is actually open to extension already. So the left hand side where you put `require` can be _any_ function.

```
(module my.module
  {require-macros {_ my.macro.module}}
```

So that should already work? I think? I'm just using an underscore to assign the result of the require to nothing, maybe it could be tweaked so you could pass a sequential table of things you want required but not `let` into a variable.

The annoying thing with Lua is there's no nice concrete way to tell the difference between a sequential and associative table, it's best to just accept one kind of table and always assume it'll be that.

**Comment by Olical** [comment]
Ah, doesn't seem to work, require-macros probably needs to be completely top level or something, not sure if I can provide that through my macro :thinking:

**Comment by Olical** [comment]
It's taken me so long to work out, but I think I've done it! You can use require-macros (and any other side effecting function or macro) from the module macro now! (on develop)

```
(module aniseed.fs
  {require {nvim aniseed.nvim}
   require-macros [fnl.aniseed.mafs]})

(add 10 20)
```

Where the `mafs` module is:

```
(fn add [a b]
  `(+ ,a ,b))

{:add add}
```

I still need to document it, but if you have a use case you'd like to test I'd love to know if it works for you :smile:

**Comment by Olical** [comment]
(pushed some docs)

**Comment by Olical** [comment]
Should be addressed in `v3.6.0` :tada:
